libsolidity/ast/Types.cpp

```diff
@@ -115,51 +115,52 @@ u256 const& MemberList::storageSize() const
 	return m_storageOffsets->storageSize();
 }
 
-TypePointer Type::fromElementaryTypeName(Token::Value _typeToken)
+TypePointer Type::fromElementaryTypeName(ElementaryTypeNameToken const& _type)
 {
-	char const* tokenCstr = Token::toString(_typeToken);
-	solAssert(Token::isElementaryTypeName(_typeToken),
-		"Expected an elementary type name but got " + ((tokenCstr) ? std::string(Token::toString(_typeToken)) : ""));
+	string tokenString = _type.toString();
+	solAssert(Token::isElementaryTypeName(_type.returnTok()),
+		"Expected an elementary type name but got " + tokenString);
 
-	if (Token::Int <= _typeToken && _typeToken <= Token::Bytes32)
+	Token::Value token = _type.returnTok();
+	unsigned int M = _type.firstNumber();
```

Comment:
could you please rename the variable? Uppercase is not allowed by coding standards